Lib/subprocess.py

```diff
@@ -1951,20 +1955,16 @@ def _execute_child(self, args, executable, preexec_fn, close_fds,
                 raise child_exception_type(err_msg)
 
 
-        def _handle_exitstatus(self, sts,
-                               _waitstatus_to_exitcode=_waitstatus_to_exitcode,
-                               _WIFSTOPPED=_WIFSTOPPED,
-                               _WSTOPSIG=_WSTOPSIG):
+        def _handle_exitstatus(self, sts, _del_safe=_del_safe):
```

Comment:
My first thought: does a `class` actually work for this purpose? If we can't use these from module globals because some finalization code may have started clearing globals, what's to prevent similar code from clearing `class _del_safe` of attributes?

The dangers of execution during `__del__` are not things i'm very familiar with.  IIRC these were added a _long_ time ago based on odd uncatchable exceptions left on stderr during shutdown.

Q1: is this old practice still needed at all in modern Python?  
Q2: if it is, is this class approach actually guaranteed to work?

Reply:
FWIW - the del-safe symbol handling is *relatively* recent - it only was added in #95181, and backported to Py3.11. 

As for whether `class` is suitable for purpose here - I hadn't considered the possible lifecycle implications. It passes the current test suite... but I'm not sure how much the test suite is exercising this particular edge case; based on the PR that introduced it, possibly not much at all.

Reply:
> the del-safe symbol handling is relatively recent - it only was added in https://github.com/python/cpython/pull/95181, and backported to Py3.11.

That was just a refactoring. We're looking at 2010's https://github.com/python/cpython/commit/42a0ba7b2c88b181bce603a57c69a66539dab693

Reply:
Testing GC state timing based `__del__` behavior at finalization is notoriously hard - so nothing explicitly covers this kind of thing.  it's a "keep beating on it until stderr noise stops" in whatever place it pops up in kind of defensive coding.  Which sadly leaves us without a good way to know if we still need it.

Reply:
Thanks - I missed that history. Not sure why the pre-refactor version didn't cause issues for me with my iOS patches.

As for whether modern Python still needs this? I assumed that if someone went to the effort of refactoring it in the 3.12 timeframe, it's still relevant - but I'll admit that's me leaning on history, rather than any deep knowledge of the problem involved.